tests/lib/bin/version-tool

```diff
@@ -0,0 +1,413 @@
+        self.assertEqual(naive_version_cmp("1-foo.2-froz", "1-bar"), +1)
+        self.assertEqual(naive_version_cmp("1-foo.2-froz", "1-bar.3-quux"), -1)
+
+    def test_kernel_version(self):
+        # type: () -> None
+        self.assertEqual(naive_version_cmp("5.2.9-1-default", "4.13"), +1)
+
+
+def strict_version_cmp(a, b):
+    # type: (Text, Text) -> int
+    """
+    strictly_version_cmp compares two version numbers without leeway.
+
+    The algorithm considers each version to be a tuple of integers. Non,
+    integer elements or element fragments are regarded as an error and raised
+    as ValueError.
+
+    Comparison is performed on by considering the leftmost element in each
+    tuple. First pair of numbers that are not equal determine the result of the
+    comparison. Tuples have unequal length then missing elements are
+    substituted with zero.
+
+    The return value is 0 if the version strings are equal, -1 if version a is
+    smaller or +1 if version b is smaller.
+    """
+    try:
+        a_items = [int(item, 10) for item in a.split(".")]
+    except ValueError:
+        raise ValueError("version {} is not purely numeric".format(a))
+    try:
+        b_items = [int(item, 10) for item in b.split(".")]
+    except ValueError:
+        raise ValueError("version {} is not purely numeric".format(b))
+    if PY2:
+        zip_longest_fn = itertools.izip_longest
+    else:
+        zip_longest_fn = itertools.zip_longest
+    for a_val, b_val in zip_longest_fn(a_items, b_items, fillvalue=0):
+        delta = a_val - b_val
+        if delta != 0:
+            return 1 if delta > 0 else -1
+    return 0
+
+
+class StrictVersionCmpTests(unittest.TestCase):
+    def test_simple(self):
+        # type: () -> None
+        self.assertEqual(strict_version_cmp("10", "10"), 0)
+        self.assertEqual(strict_version_cmp("10", "20"), -1)
+        self.assertEqual(strict_version_cmp("20", "10"), +1)
+
+    def test_many_segments(self):
+        # type: () -> None
+        self.assertEqual(strict_version_cmp("1.2.3", "1.2.3"), 0)
+        self.assertEqual(strict_version_cmp("1.2.3", "1.3.4"), -1)
+        self.assertEqual(strict_version_cmp("1.4.3", "1.2.3"), +1)
+        self.assertEqual(strict_version_cmp("1.0.0", "1.1.0"), -1)
+        self.assertEqual(strict_version_cmp("0.1.2", "1.1.2"), -1)
+
+    def test_unequal_length(self):
+        # type: () -> None
+        self.assertEqual(strict_version_cmp("1", "1.0"), 0)
+        self.assertEqual(strict_version_cmp("1", "1.2"), -1)
+        self.assertEqual(strict_version_cmp("1.2", "1"), +1)
+        self.assertEqual(strict_version_cmp("1", "1.0.1"), -1)
+        self.assertEqual(strict_version_cmp("1.1", "1.0.1"), +1)
+
+    def test_version_with_text(self):
+        # type: () -> None
+        with self.assertRaises(ValueError) as cm:
+            strict_version_cmp("1-foo", "1")
+        self.assertEqual(cm.exception.message, "version 1-foo is not purely numeric")
```

Comment:
`str(cm.exception)` IIRC message is Python2 only

Reply:
I fixed this, shucks.